dts/arm/nordic/nrf52840.dtsi

```diff
@@ -537,16 +537,11 @@
 		};
 
 		cryptocell: crypto@5002a000 {
-			compatible = "nordic,nrf-cc310";
-			reg = <0x5002A000 0x1000>;
-			status = "okay";
-			#address-cells = <1>;
-			#size-cells = <1>;
-			cryptocell310: crypto@5002b000 {
-				compatible = "arm,cryptocell-310";
-				reg = <0x5002B000 0x1000>;
-				interrupts = <42 NRF_DEFAULT_IRQ_PRIORITY>;
-			};
+			compatible = "nordic,cryptocell", "arm,cryptocell-310";
+			reg = <0x5002a000 0x1000>, <0x5002b000 0x1000>;
+			reg-names = "wrapper", "core";
+			interrupts = <42 NRF_DEFAULT_IRQ_PRIORITY>;
+			status = "disabled";
```

Comment:
This lgtm, but I would suggest having just one compatible here, like before.

Having multiple compatibles influences driver selection. If there existed a generic driver for `arm,cryptocell-310`, then I'd assume it should be able to drive this peripheral using the non-Nordic set of registers. That would probably require putting the "core" registers aside into a separate node. If that is not possible, then this peripheral should only be Nordic compatible.

This is purely theoretical, and I'm not familiar with CryptoCell, so I don't know if that makes sense.

Reply:
having one compatible doesn't follow devicetree spec, IMHO. 

> The compatible property value consists of one or more strings that define the specific programming model for
the device. This list of strings should be used by a client program for device driver selection. The property
value consists of a concatenated list of null terminated strings, from most specific to most general. They allow
a device to express its compatibility with a family of similar devices, potentially allowing a single device driver
to match against several devices.

The only specific bit of this IP integration is the extra Nordic wrapper used to enable it. So in theory, a single vendor-agnostic driver could be implemented, provided it supports vendor quirks (wrapper in this case).

Reply:
> a single vendor-agnostic driver could be implemented, provided it supports vendor quirks (wrapper in this case).

That's what I was unsure about, but if this is acceptable then I'm OK with it :+1: